src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,92 @@
+
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository)
+    {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        if (!$this->isPromotionApplied($channelPricing)) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $latestLogEntry = $this->getLatestLogEntry($channelPricing);
+        $lowestPriceInPeriod = $this->findLowestPriceInPeriod($latestLogEntry);
+
+        if ($lowestPriceInPeriod !== null) {
+            $channelPricing->setLowestPriceBeforeDiscount($lowestPriceInPeriod);
+
+            return;
+        }
+
+        $latestPriceBeyondPeriod = $this->findLatestPriceBeyondPeriod($latestLogEntry);
+
+        $channelPricing->setLowestPriceBeforeDiscount($latestPriceBeyondPeriod);
+    }
+
+    private function isPromotionApplied(ChannelPricingInterface $channelPricing): bool
+    {
+        return
+            $channelPricing->getOriginalPrice() !== null &&
+            $channelPricing->getPrice() < $channelPricing->getOriginalPrice()
+        ;
+    }
+
+    private function getLatestLogEntry(ChannelPricingInterface $channelPricing): ChannelPricingLogEntryInterface
+    {
+        $latestLogEntry = $this->channelPricingLogEntryRepository->findLatestOne($channelPricing);
+        Assert::notNull($latestLogEntry);
+
+        return $latestLogEntry;
+    }
+
+    private function findLowestPriceInPeriod(ChannelPricingLogEntryInterface $latestLogEntry): ?int
+    {
+        $loggedAt = new \DateTimeImmutable($latestLogEntry->getLoggedAt()->format('Y-m-d H:i:s'));
+
+        /** @var \DateTimeInterface $startDate */
+        $startDate = $loggedAt->sub(new \DateInterval(sprintf('P%sD', self::PERIOD_IN_DAYS)));
+
+        return $this
+            ->channelPricingLogEntryRepository
+            ->findLowestPriceInPeriod($latestLogEntry->getId(), $latestLogEntry->getChannelPricing(), $startDate)
+        ;
+    }
+
+    private function findLatestPriceBeyondPeriod(ChannelPricingLogEntryInterface $latestLogEntry): ?int
+    {
+        $loggedAt = new \DateTimeImmutable($latestLogEntry->getLoggedAt()->format('Y-m-d H:i:s'));
+
+        /** @var \DateTimeInterface $startDate */
+        $startDate = $loggedAt->sub(new \DateInterval(sprintf('P%sD', self::PERIOD_IN_DAYS)));
+
+        return $this
+            ->channelPricingLogEntryRepository
+            ->findLatestPriceBeyondPeriod($latestLogEntry->getChannelPricing(), $startDate)
+        ;
+    }
```

Comment:
I thought for a while about what's wrong with this method, but it's a bit misleading, because we're not looking for the latest price outside the period, we're actually looking for the latest price, which is also the price that was in force during the period. IMO, this price should be returned in the `findLowestPriceInPeriod` method, as it should return the lowest price of the period, not the lowest price from the log of the period, and this additional method on the repository `findLatestPriceBeyondPeriod` should not be needed.